docker/spack-stack/Dockerfile

```diff
@@ -0,0 +1,534 @@
+&&   echo '      version: [1.2.0]' \
+&&   echo '    yafyaml:' \
+&&   echo '      version: [0.5.1]' \
+&&   echo '    zlib:' \
+&&   echo '      version: [1.2.13]' \
+&&   echo '    zstd:' \
+&&   echo '      version: [1.5.2]' \
+&&   echo '      variants: +programs' \
+&&   echo '' \
+&&   echo '  specs: [base-env@1.0.0, jedi-base-env@1.0.0, ewok-env@1.0.0, jedi-fv3-env@1.0.0,' \
+&&   echo '    jedi-mpas-env@1.0.0, bacio@2.4.1, bison@3.8.2, bufr@12.0.0, ecbuild@3.7.2, eccodes@2.27.0,' \
+&&   echo '    ecflow@5, eckit@1.24.4, ecmwf-atlas@0.35.0 +fckit +ectrans +tesselation +fftw,' \
+&&   echo '    fiat@1.2.0, ectrans@1.2.0 +fftw, eigen@3.4.0, fckit@0.11.0, fms@release-jcsda,' \
+&&   echo '    g2@3.4.5, g2tmpl@1.10.0, gftl-shared@1.6.1, gsibec@1.1.3, hdf@4.2.15, hdf5@1.14.0,' \
+&&   echo '    ip@4.3.0, jasper@2.0.32, jedi-cmake@1.4.0, libpng@1.6.37, nccmp@1.9.0.1, netcdf-c@4.9.2,' \
+&&   echo '    netcdf-cxx4@4.3.1, netcdf-fortran@4.6.0, nlohmann-json@3.10.5, nlohmann-json-schema-validator@2.1.0,' \
+&&   echo '    parallelio@2.5.10, parallel-netcdf@1.12.2, py-eccodes@1.4.2, py-f90nml@1.4.3,' \
+&&   echo '    py-gitpython@3.1.27, py-h5py@3.7.0, py-numpy@1.22.3, py-pandas@1.5.3, py-pip,' \
+&&   echo '    py-pyyaml@5.4.1, py-scipy@1.9.3, py-shapely@1.8.0, py-xarray@2022.3.0, sp@2.3.3,' \
+&&   echo '    udunits@2.2.28, w3nco@2.4.1, w3emc@2.10.0, nco@5.0.6, esmf@8.5.0, mapl@2.40.3,' \
+&&   echo '    zlib@1.2.13, zstd@1.5.2, odc@1.4.6, shumlib@macos_clang_linux_intel_port, awscli@1.27.84,' \
+&&   echo '    py-globus-cli@3.16.0]' \
+&&   echo '  include: []') > /opt/spack-environment/spack.yaml
+
+# Install the software, remove unnecessary deps
+RUN cd /opt/spack-environment && spack env activate . && spack install --fail-fast && spack gc -y
+
+# Modifications to the environment that are necessary to run
+RUN cd /opt/spack-environment && \
+    spack env activate --sh -d . > activate.sh
+
+# Put output of spack find into a file
+RUN cd /opt/spack-environment && \
+spack env activate -d . && \
+spack find 2>&1 | tee /root/spack_find.out
+
+
+# Bare OS image to run the installed executables
+FROM ubuntu:20.04
+
+COPY --from=builder /opt/spack-environment /opt/spack-environment
+COPY --from=builder /opt/software /opt/software
+COPY --from=builder /opt/._view /opt/._view
+COPY --from=builder /opt/view /opt/view
+
+RUN { \
+      echo '#!/bin/sh' \
+      && echo '.' /opt/spack-environment/activate.sh \
+      && echo 'exec "$@"'; \
+    } > /entrypoint.sh \
+&&  chmod a+x /entrypoint.sh
+
+
+#Set environment variables for installing tzdata
+ENV DEBIAN_FRONTEND=noninteractive
+ENV TZ=Etc/UTC
+ENV CC=gcc
+ENV CXX=g++
+ENV FC=gfortran
+
+
+RUN apt-get -yqq update && apt-get -yqq upgrade \
+ && apt-get -yqq install bc cpp g++ gcc gfortran git git-lfs make mysql-server qt5-default libqt5svg5-dev qt5dxcb-plugin wget build-essential ca-certificates curl file gnupg2 iproute2 locales python3 python3-pip python3-setuptools unzip vim \
```

Comment:
Same here with these lines. Consider breaking up into multiple lines for clearly more consistent look.